build.sbt

```diff
@@ -1,125 +1,11 @@
-    settings(toolSettings)
-
-lazy val nir =
-  project.in(file("nir")).
-    settings(toolSettings).
-    dependsOn(util)
-
-lazy val tools =
-  project.in(file("tools")).
-    settings(toolSettings).
-    settings(
-      libraryDependencies += "commons-io" % "commons-io" % "2.4"
-    ).
-    dependsOn(nir, util)
-
-lazy val nscplugin =
-  project.in(file("nscplugin")).
-    settings(toolSettings).
-    settings(
-      scalaVersion := "2.11.8",
-      unmanagedSourceDirectories in Compile ++= Seq(
-        (scalaSource in (nir, Compile)).value,
-        (scalaSource in (util, Compile)).value
-      ),
-      libraryDependencies ++= Seq(
-        "org.scala-lang" % "scala-compiler" % scalaVersion.value,
-        "org.scala-lang" % "scala-reflect" % scalaVersion.value
-      ),
-      publishArtifact in (Compile, packageDoc) := false
-    )
-
-lazy val sbtplugin =
-  project.in(file("sbtplugin")).
-    settings(toolSettings).
-    settings(
-      sbtPlugin := true
-    ).
-    dependsOn(tools)
-
-// rt is a library project but it can't use libSettings
-// due to the fact that it contains nrt dependency in
-// ScalaNativePlugin.projectSettings.
-lazy val rtlib =
-  project.in(file("rtlib")).
-    settings(baseSettings).
-    settings(
-      scalaVersion := libScalaVersion
-    )
-
-lazy val nativelib =
-  project.in(file("nativelib")).
-    settings(libSettings)
-
-lazy val javalib =
-  project.in(file("javalib")).
-    settings(libSettings).
-    dependsOn(nativelib)
-
-lazy val assembleScalaLibrary = taskKey[Unit](
-  "Checks out scala standard library from submodules/scala and then applies overrides.")
-
-lazy val scalalib =
-  project.in(file("scalalib")).
-    settings(libSettings).
-    settings(
-      assembleScalaLibrary := {
-        IO.delete(file("scalalib/src/main/scala"))
-        IO.copyDirectory(
-          file("submodules/scala/src/library/scala"),
-          file("scalalib/src/main/scala/scala"))
-
-        val epoch :: major :: _ = scalaVersion.value.split("\\.").toList
-        IO.copyDirectory(
-          file(s"scalalib/overrides-$epoch.$major/scala"),
-          file("scalalib/src/main/scala/scala"), overwrite = true)
-      },
-
-      compile in Compile <<= (compile in Compile) dependsOn assembleScalaLibrary
-    ).
-    dependsOn(javalib)
-
-lazy val demoNative =
-  project.in(file("demo-native")).
-    settings(libSettings).
-    settings(
-      nativeVerbose := true,
-      nativeClangOptions := Seq("-O2")
-    ).
-    dependsOn(scalalib)
-
-lazy val demoJVM =
-  project.in(file("demo-jvm")).
-    settings(
-      fork in run := true,
-      javaOptions in run ++= Seq("-Xms64m", "-Xmx64m")
-    )
+val util = Build.util
+val nir = Build.nir
+val tools = Build.tools
+val nscplugin = Build.nscplugin
+val sbtplugin = Build.sbtplugin
+val rtlib = Build.rtlib
+val nativelib = Build.nativelib
+val javalib = Build.javalib
+val scalalib = Build.scalalib
+val demoNative = Build.demoNative
+val demoJVM = Build.demoJVM
```

Comment:
Tests with Build.scala is more readable.
But I find this https://github.com/sbt/sbt/pull/2524.
Using build.sbt is better now.